plugins/jira/unit_test/test_create_issue.py

```diff
@@ -0,0 +1,166 @@
+from collections import namedtuple
+from unit_test.payloads.client_fields import client_fields
+
+######################
+# MOCKS
+######################
+
+"""
+This is a difficult test. The Jira API returns very big, very custom objects. To unit test the api
+requires us to recreate those objects the best we can. Therefore there's a lot of mocks in this test
+
+You'll see a lot of this: 
+
+namedtuple("AnObject", ["name"])(["new years"]),
+
+This will create a faux object that can be accessed like this: 
+
+AnObject.name
+
+That code can be used to quickly convert a dict to a fake object. 
+"""
+
+
+class MockIssue:
+    def __init__(self):
+        fields_dict = {
+            "resolution": namedtuple("AnObject", ["name"])(["new years"]),
+            "reporter": namedtuple("AnObject", ["displayName"])(["Bob Smith"]),
+            "assignee": namedtuple("AnObject", ["displayName"])(["Bob Smith"]),
+            "resolutiondate": "No idea what this is",
+            "description": "A description",
+            "summary": "A summary",
+            "status": namedtuple("AnObject", ["name"])(["In Progress"]),
+            "created": "Yesterday",
+            "updated": "Yesterday",
+            "labels": ["blocked"],
+        }
+
+        self.raw = {"fields": "something"}
+        self.id = "12345"
+        self.key = "12345"
+        self.fields = namedtuple("ObjectName", fields_dict.keys())(*fields_dict.values())
+
+    def permalink(self):
+        return "https://example-demo.atlassian.net/browse/ISSUE-ID-1234"
+
+
+class MockClient:
+    def __init__(self):
+        self.client = "some fake thing"
+
+    def projects(self):
+        project = {
+            "raw": {},
+            "expand": "description,lead,issueTypes,url,projectKeys,permissions,insight",
+            "self": "https://example.atlassian.net/rest/api/2/project/12345",
+            "id": "12345",
+            "key": "projectKey",
+            "name": "projectName",
+            "avatarUrls": {},
+            "entityId": "12345-12345-12345-12345",
+            "uuid": "12345-12345-12345-12345",
+        }
+
+        # Need to convert this to an object to simulate the JiraObject return
+        project_object = namedtuple("ObjectName", project.keys())(*project.values())
+
+        return [project_object]
+
+    def create_issue(self, fields):
+        return MockIssue()
+
+    def issue_type_by_name(self, issue_type):
+        pass
+
+    def fields(self):
+        return client_fields
+
+
+class MockConnection:
+    def __init__(self):
+        self.client = MockClient()
+
+
+######################
+# Tests
+######################
+
+
+class TestCreateIssue(TestCase):
+    def setUp(self):
+        self.test_conn = Connection()
+        self.test_action = CreateIssue()
+
+        test_logger = logging.getLogger("test")
+        self.test_conn.logger = test_logger
+        self.test_action.logger = test_logger
+
+    def test_create_issue(self):
```

Comment:
Can you add additional tests for:
- Issue without a type
- Issue without a description
- Totally bizarre project ID (examples: symbols, upper/lower-case mix, all numbers, etc.)

Jira is a heavy hitter plugin and we get a lot of issues for it, so it would be good to flesh out the test cases a bit more even if it means we need to spend more time writing them. Let me know what you think around this.

Reply:
Yup, I wasn't done with this PR. I opened it so I could see my committed code and see where I'm at. 

I didn't add [WIP] my bad.